src/CommandingOfficers/DeployableCommander.java

```diff
@@ -0,0 +1,263 @@
+  public ArrayList<Unit> COUs = new ArrayList<Unit>();
+  public ArrayList<Unit> COUsLost = new ArrayList<Unit>();
+  public final DeployCOUFactory deployAction; // Has to be one per CO, so units get the same instance over multiple action searches (mostly for the AI)
+  /** The number of COUs you can have active at once */
+  public abstract int getCOUCount();
+  public void onCOULost(Unit minion) {};
+  public char getCOUMark() {return 'C';};
+  public int deployCostPercent = 0;
+  /** If the unit type matches any flag in this mask, it can be my COU */
+  public long canDeployMask = Long.MAX_VALUE;
+  public final boolean canDeployOn(UnitModel type) {return type.isAny(canDeployMask);};
+
+  public boolean resetCOUsEveryTurn = false;
+
+  /** Can I deploy my COU here? Ignored if resetCOUsEveryTurn==true */
+  protected boolean eligibleDeployLocation(Unit actor, MapLocation loc)
+  {
+    if( loc.getOwner() != actor.CO )
+      return false; // No COing on neutral factories
+    return getShoppingList(loc).contains(actor.model)
+        || loc.getEnvironment().terrainType == TerrainType.HEADQUARTERS
+        || loc.getEnvironment().terrainType == TerrainType.LAB;
+  }
+
+
+  public DeployableCommander(CommanderInfo info, GameScenario.GameRules rules)
+  {
+    super(info, rules);
+    deployAction = new DeployCOUFactory(this);
+  }
+
+  @Override
+  public char getUnitMarking(Unit unit, Army activeArmy)
+  {
+    if( COUs.contains(unit) )
+      return getCOUMark();
+
+    return super.getUnitMarking(unit, activeArmy);
+  }
+
+  @Override
+  public GameEventQueue initTurn(MapMaster map)
+  {
+    if( resetCOUsEveryTurn )
+    {
+      this.COUs.clear();
+    }
+    else
+    {
+      this.COUs.removeAll(COUsLost);
+      this.COUsLost.clear();
+    }
+    return super.initTurn(map);
+  }
+
+  @Override
+  public void endTurn()
+  {
+    super.endTurn();
+  }
+
+  @Override
+  public void modifyActionList(UnitContext uc)
+  {
+    if( canDeployOn(uc.model) )
+      uc.actionTypes.add(deployAction);
+  }
+
+  @Override
+  public GameEventQueue receiveUnitDieEvent(Unit victim, XYCoord grave, Integer hpBeforeDeath)
+  {
+    // COUs die when they are killed
+    if( COUs.contains(victim) )
+    {
+      COUsLost.add(victim);
+      onCOULost(victim);
+    }
+    return super.receiveUnitDieEvent(victim, grave, hpBeforeDeath);
+  }
+  @Override
+  public GameEventQueue receiveUnitJoinEvent(JoinEvent join)
+  {
+    // COUs shouldn't vanish into the aether on joining another unit
+    if( COUs.contains(join.unitDonor) )
+    {
+      COUs.remove(join.unitDonor);
+      COUs.add(join.unitRecipient);
+    }
+    return super.receiveUnitJoinEvent(join);
+  }
+
+  //////////////////////////////////////////////////////////
+  // Action definition happens after this point
+  //////////////////////////////////////////////////////////
+
+  private static class DeployCOUFactory extends UnitActionFactory
```

Comment:
Do we want to keep spreading action/factory/event classes everywhere (especially given that we have dedicated packages for those things already? I mean there's global search, but if I don't remember the proper name for a given thing it may not help much.

Reply:
I wasn't under the impression they were being spread "everywhere".
This is code for this specific subtype of CO, and isn't applicable to anyone else. Ditto with Meridian.
I also don't foresee other pieces of code needing/wanting to use the Kaiju actions for anything - they're uh, very unique (and only work if every unit around is from the KW set).

It seems more logical to me for this code to be put with the CO/unit set that uses them, rather than in the general "actions for units" bin.
IIRC I moved some of the transform stuff out of Meridian when it started being used by subs/stealths.
I suppose I don't mind relocating this code super strongly, but I would mind it a little bit.

Reply:
Fair enough.